python/pyspark/mllib/linalg.py

```diff
@@ -634,6 +634,18 @@ def toArray(self):
         """
         return self.values.reshape((self.numRows, self.numCols), order='F')
 
+    def __getitem__(self, indices):
+        i, j = indices
+        if i > self.numRows or i < 0:
```

Comment:
Sorry, I didn't notice this in the first pass. The bound should be `[0, numRows)`:

~~~
if i < 0 or i >= self.numRows:
  raise ValueError("Row index %d is out of range range [0, %d)" % (i, self.numRows))
~~~

Same for `j` below.